m2m-transfers/V3/REPWRITERSPECS/BANNO.M2MTRANSFERS.RPT.BATCH

```diff
@@ -0,0 +1,751 @@
+        END
+      END
+     ELSE [transaction is not a comment - validate & process]
+      DO
+       IF TCCOUNT=1 THEN [transaction with statement description]
+        DO
+         IF CHARACTERSEARCH(LOAN TRANSACTION:STMTDESC:1,TARGETCOMMENT)>0 THEN
+          DO [valid target transaction - add to report]
+           BNOTEXT=SEGMENT(TCCOMMENT(TCCOUNT),4,LENGTH(TCCOMMENT(TCCOUNT)))
+           CALL BNONLTS
+           BNODELIMITER=BNODELIMSPACE
+           BNODATALINEINPUT=BNOTEXT
+           CALL PARSEDATALINE
+           TOACCOUNT=BNODATALINEFIELD(BNODATAFIELDCOUNT-2)
+           TOSLIDTYPE=BNODATALINEFIELD(BNODATAFIELDCOUNT-1)
+           TOSLID=BNODATALINEFIELD(BNODATAFIELDCOUNT)
+           TOTRANAMOUNT=LOAN TRANSACTION:TRANAMOUNT
+           CALL PRINTTRANTOREPORT
+           CALL CLEARTRANVARS
+          END
+        END [IF TCCOUNT=1 THEN...]
+       ELSE IF TCCOUNT=2 THEN [transaction with tran comment]
+        DO
+         IF CHARACTERSEARCH(TCCOMMENT(TCCOUNT-1),TARGETCOMMENT)>0 THEN
+          DO
+           BNOTEXT=SEGMENT(TCCOMMENT(TCCOUNT),4,LENGTH(TCCOMMENT(TCCOUNT)))
+           CALL BNONLTS
+           BNODELIMITER=BNODELIMSPACE
+           BNODATALINEINPUT=BNOTEXT
+           CALL PARSEDATALINE
+
+           TOACCOUNT=BNODATALINEFIELD(BNODATAFIELDCOUNT-2)
+           TOSLIDTYPE=BNODATALINEFIELD(BNODATAFIELDCOUNT-1)
+           TOSLID=BNODATALINEFIELD(BNODATAFIELDCOUNT)
+           TOTRANAMOUNT=LOAN TRANSACTION:TRANAMOUNT
+           CALL PRINTTRANTOREPORT
+           CALL CLEARTRANVARS
+          END [ELSE IF TCCOUNT=2 THEN...]
+        END [IF TCCOUNT=2]
+      END [LOAN TRANSACTION:COMMENTCODE<>1]
+    END [FOR EACH LOAN TRANSACTION WITH...]
+  END [FOR EACH LOAN]
+END [PROCEDURE]
+
+PROCEDURE PRINTTRANTOREPORT
+[* Print the detail for an immediate transfer (from tran history)
+*]
+ COL=001 ACCOUNT:NUMBER
+ COL=013 TRANSOURCECHR(TRANSOURCE)
+ COL=020 TRANSOURCEID
+ COL=026 TOACCOUNT
+ COL=038 TOSLIDTYPE
+ COL=045 TOSLID
+ COL=061 RIGHT TOTRANAMOUNT
+ COL=064 "IMMEDIATE"
+ NEWLINE
+END
+
+PROCEDURE CLEARTRANVARS
+[* Clear the variables storing transactional detail
+*]
+ TCCOUNT=0
+ FOR TMPLOOP=1 TO TCCOUNT
+  DO
+   TCCOMMENT(TMPLOOP)=""
+   TCSTMTDESCR(TMPLOOP)=""
+   TCIDTYPE(TMPLOOP)=0
+   TCID(TMPLOOP)=""
+  END
+ TOACCOUNT=""
+ TOSLIDTYPE=""
+ TOSLID=""
+ TOTRANAMOUNT=$0.00
+END [PROCEDURE]
+
+PROCEDURE GETSHARETRANSFERRECS
+[* Look through FM history for Share transfer records created within the
+** desired date range and by the desired user number
+*]
+ CALL CLEARFIELDS
+
+ FOR EACH FMHISTORY WITH (FMHISTORY:POSTDATE>=TARGETSTARTDATE AND
+                          FMHISTORY:POSTDATE<=TARGETENDDATE)
+  DO
+   IF FMHISTORY:FMTYPE=0 AND
+      FMHISTORY:RECORDTYPE=8 AND [SHARE TRANSFER RECORD]
+      FMHISTORY:USERNUMBER=TARGETSYMXUSER AND
+      CHARACTERSEARCH(UPPERCASE(FMHISTORY:FIELDNAME),"NEW LOCATOR")>0 THEN
+    DO
+     NEWLOCFOUND=TRUE
+     TARGETSEQUENCE=FMHISTORY:SEQUENCENUMBER
```

Comment:
Hi Tom, First I like the code you wrote here. It is very solid and well reinforced for the standard situation.
That said I found out the hard way a CU can have FM history for a New Locator going in one direction for the Live SYM and then another direction for the UAT SYM, thereby break Logic that assumes the first field will be the New Locator for the new Sequence number of the FM Record Type you are looking for. I think this will need to be addressed in this code before general release to all CU's as if a UAT can be reversed in order for the fields in an FM History Record, then so can a Live SYM.

Here is the location of the code where I have recently delt with such a problem
https://github.com/Banno/custom-powerons/blob/03a18b2b4aaf5b99917f5a2af5bcf0e60be52a79/446-Transit%20Employees%20FCU/13985356%20-%20Card%20Limits/REPWRITERSPECS/BNOCUST.446.CARD.LIMITS.POW#L579

And the code itsself. You can probably do it better. The example is just one way to handle the situation.
PROCEDURE CHECKFOROVERMAXINCREASES
[* Looks at FM History to determine if the Card has
** more than Max Increases in 30 days
*]

 FOR TMPLOOP=1 TO CARDCOUNT
  DO
   CURRENTSEQNUM=0
   CARDLOCFOUND=FALSE
   INCREASEFOUND=FALSE
   INCREASESCOUNT=0
   FOR EACH FMHISTORY WITH(FMHISTORY:POSTDATE>=DATEOFFSET(SYSTEMDATE,-1,0) AND
                           FMHISTORY:FMTYPE=1 AND
                           FMHISTORY:RECORDTYPE=19 AND
                           PARAMIEXEMPTUSERS(FMHISTORY:USERNUMBER)=FALSE)
    DO
     IF CURRENTSEQNUM<>FMHISTORY:SEQUENCENUMBER THEN [reset variables when sequence # changes]
      DO
       CURRENTSEQNUM=FMHISTORY:SEQUENCENUMBER
       CARDLOCFOUND=FALSE
       INCREASEFOUND=FALSE
      END
     [set card loc found flag if limit increased]
     IF UPPERCASE(FMHISTORY:FIELDNAME)="REVISE CARD LOC" AND
        FMHISTORY:NEWNUMBER=CARDLOCATOR(TMPLOOP) THEN
         CARDLOCFOUND=TRUE

     [set increase found flag if limit increased]
     ELSE IF UPPERCASE(FMHISTORY:FIELDNAME)="POS LIMIT" AND
             FMHISTORY:OLDMONEY<FMHISTORY:NEWMONEY THEN
              INCREASEFOUND=TRUE
     [evaluate card over max increases if card loc flag found and if limit increased]
     IF CARDLOCFOUND=TRUE AND INCREASEFOUND=TRUE THEN
      DO
       INCREASESCOUNT=INCREASESCOUNT+1 [Increment if limit increased & if eligible card locator]
       IF INCREASESCOUNT>PARAMMAXINCREASES AND PARAMMAXINCREASES>0 THEN
        CARDOVERMAXINCREASES(TMPLOOP)=TRUE [if over max increases, set flag]
      END
    END
   UNTIL FMHISTORY:POSTDATE<DATEOFFSET(SYSTEMDATE,-1,0) OR CARDOVERMAXINCREASES(TMPLOOP)=TRUE
  END [TMPLOOP=1 TO CARDCOUNT]
END

Reply:
@jordan-keenan I'll quote Justin here... "It's been 'today' number of days when I learned.....! Or, something like that!  In any case when you have time ping me so we can discuss your comment pertaining the FM history direction.

Reply:
Modified procedure to somewhat mirror the logic you posted here